### ISSUE TITLE
Enforce correct linebreaks in JSX

### ISSUE THREAD

**Issue by mattheu** [humanmade/coding-standards#171]
We should add the rule `react/jsx-curly-newline` to ensure newlines are used consistently in JSX.

The following is allowed

```jsx
/* eslint-disable no-unused-vars */

import React from 'react';

const foo = 'foo';
const bar = 'bar';

const A = () => (
	<div>
		{ foo }
	</div>
);

const B = () => (
	<div>
		{
			foo
		}
	</div>
);

```

But the following is not

```jsx
/* eslint-disable no-unused-vars */

import React from 'react';

const foo = 'foo';

const A = () => (
	<div>
		{ foo
		}
	</div>
);

const B = () => (
	<div>
		{
			foo }
	</div>
);

const C = () => (
	<div>
		{ foo &&
			foo.bar
		}
	</div>
);
```

**Comment by rmccue** [comment]
This would presumably still allow the following, right?

```jsx
<div>
    { foo && (
        foo.bar
    ) }

    { foo ? (
        foo.bar
    ) : null }

    { foo ? (
        foo.bar
    ) : bar }

    { foo ? (
        foo.bar
    ) : (
        bar
    ) }
</div>
```

**Comment by mattheu** [comment]
Yes all of those are still allowed - i can add that exact code to the `pass` tests
